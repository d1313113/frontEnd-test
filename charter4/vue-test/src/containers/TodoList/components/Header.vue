<template>
  <div class="header">
    <div class="header-content">
      <input type="text" data-test="input" v-model="inputValue" @keyup.enter="addTodoItem">
      <input type="text" data-test="input1" :value="inputValueVuex" @input="e => changeInputValue(e.target.value)" @keyup.enter="addTodoItem">
    </div>
  </div>
</template>

<script>
import { mapState, mapMutations } from 'vuex'
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  computed: {
    ...mapState({
      inputValueVuex: state => state.inputValueVuex
    })
  },
  methods: {
    addTodoItem () {
      if (this.inputValue || this.inputValueVuex) {
        this.$emit('add', this.inputValue || this.inputValueVuex)
        this.inputValue = ''
        this.changeInputValue('')
      }
    },
    ...mapMutations({
      changeInputValue: 'changeInputValue'
    })
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
.header
  background-color #666
  line-height 60px
  .header-content
    width 600px
    margin 0 auto
    color #fff
    font-size 24px
</style>
