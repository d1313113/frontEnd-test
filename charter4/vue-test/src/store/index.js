import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

const store = new Vuex.Store({
  state: {
    inputValueVuex: 'cumelmell'
  },
  mutations: {
    changeInputValue (state, payload) {
      state.inputValueVuex = payload
    }
  }
})

export default store
