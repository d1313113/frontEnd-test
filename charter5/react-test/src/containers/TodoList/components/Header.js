import React, { Component } from 'react';

class Header extends Component {

  constructor(props) {
    super(props);
    this.state = {
      value: ''
    }
  }

  handleInputChange = (e) => {
    this.setState({
      value: e.target.value
    });
  }

  handleKeyUp = (e) => {
    const { value } = this.state;
    if (e.keyCode === 13 && value) {
      this.props.addUndoItem(value);
      this.setState({
        value: ''
      });
    }
  }

  render() {
    const { value } = this.state;
    return (
      <div>
        <input
          type="text"
          data-test="input"
          value={ value }
          onChange={ this.handleInputChange }
          onKeyUp={ this.handleKeyUp }
        />
      </div>
    );
  }
}

export default Header;
