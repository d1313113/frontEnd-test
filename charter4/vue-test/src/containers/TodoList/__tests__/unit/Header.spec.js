import { shallowMount } from '@vue/test-utils';
import Header from '../../components/Header.vue';
import { findTestWrapper } from '../../../../utils/testUtils';
import store from '../../../../store';

describe('Header.vue', () => {

  it('Header 样式改变提示', () => {
    const wrapper = shallowMount(Header, { store });
    expect(wrapper).toMatchSnapshot();
  })

  it('Header 包含 input', () => {
    const wrapper = shallowMount(Header, { store });
    const input = findTestWrapper(wrapper, 'input');
    expect(input.exists()).toBe(true);
  });

  it('Header 中 input 框初始内容为空', () => {
    const wrapper = shallowMount(Header, { store });
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe('');
  });

  it('Header 中 input 发生变化,数据应该变化', () => {
    const wrapper = shallowMount(Header, { store });
    const input = findTestWrapper(wrapper, 'input');
    input.setValue('cumelmell');
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe('cumelmell');
  });

  it('Header 中 input 回车,无内容,无反应', () => {
    const wrapper = shallowMount(Header, { store });
    const input = findTestWrapper(wrapper, 'input');
    input.setValue('');
    input.trigger('keyup.enter');
    expect(wrapper.emitted().add).toBeTruthy();
  });

  it('Header 中 input 回车,有内容,触发 add 事件,并清空 inputValue', () => {
    const wrapper = shallowMount(Header, { store });
    const input = findTestWrapper(wrapper, 'input').at(0);
    input.setValue('aaaa');
    input.trigger('keyup.enter');
    expect(wrapper.emitted().add).toBeTruthy();
    expect(wrapper.vm.inputValue).toBe('');
  });
});
