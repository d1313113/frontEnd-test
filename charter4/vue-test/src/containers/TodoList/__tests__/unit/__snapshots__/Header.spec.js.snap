// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Header 样式改变提示 1`] = `
<div class="header">
  <div class="header-content"><input type="text" data-test="input"></div>
</div>
`;
